p2p/net/conn/dial.go

```diff
@@ -144,10 +144,24 @@ func (d *Dialer) rawConnDial(ctx context.Context, raddr ma.Multiaddr, remote pee
 		}
 	}
 
+	useLocalAddr := true
+	for _, p := range raddr.Protocols() {
+		if p.Name == "utp" {
+			useLocalAddr = false
```

Comment:
is this the lib problem? we need to fix it. in many situations people _will not_ be able to open another port. they need to be able to specify the _one port_ everything happens through.